### PR TITLE
feat(package-rules): matchNewValue

### LINKED ISSUE #27313: New option: matchNewValue
### Describe the proposed change(s).

Add a new option matchNewValue which is used to match against the newValue string, equivalent to matchCurrentValue

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
Adds new packageRules.matchNewValue

## Context
Closes: https://github.com/renovatebot/renovate/issues/27313

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository https://github.com/RahulGautamSingh-testing/match-new-value/pulls

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
Will open for review after test runs

File: lib/workers/repository/updates/flatten.ts [hunk 1]
```diff
@@ -177,6 +177,7 @@ export async function flattenUpdates(
                 config.vulnerabilityAlerts,
               );
               delete updateConfig.vulnerabilityAlerts;
+
```
Comment (RahulGautamSingh) [inline lib/workers/repository/updates/flatten.ts:180]:
```suggestion
```

**Comment by RahulGautamSingh** [comment]
Test Run: https://github.com/RahulGautamSingh-testing/match-new-value/pulls

File: lib/util/package-rules/new-value.spec.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      expect(result).toBeFalse();
+    });
+
+    it('case insensitive match', () => {
+      const result = matcher.matches(
+        {
+          newValue: '"V1.1.0"',
+        },
+        {
+          matchNewValue: '/^"v/i',
```
Comment (viceice) [inline lib/util/package-rules/new-value.spec.ts:37]: add a note about the `i` suffix for case insensitive matching

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**